src/App.js

```diff
@@ -1,24 +1,38 @@
-import logo from './logo.svg';
-import './App.css';
+import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
+import { GlobalStyle } from "./globalstyles";
+import { useState } from "react";
+import { ThemeProvider } from "styled-components";
+import { lightTheme, darkTheme } from "./components/DarkMode";
+import { ThemeChanger } from "./components/DarkMode/ThemeStyles";
+import { Button } from "./components/HeroSection/HeroSection";
+import Home from "./components/Pages/Home";
+import About from "./components/Pages/About";
+import Works from "./components/Pages/Works";
+import Navbar from "./components/Menu";
+import Footer from "./components/Footer";
 
 function App() {
+  const [theme, setTheme] = useState("light");
+
+  const themeToggler = () => {
+    theme === "light" ? setTheme("dark") : setTheme("light");
+  };
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <Router>
+      <ThemeProvider theme={theme === "light" ? lightTheme : darkTheme}>
+        <GlobalStyle />
+        <ThemeChanger>
+          <Button onClick={() => themeToggler()}> My Dark Mode </Button>
```

Comment:
Se não me engano, pode usar só:
```suggestion
onClick={themeToggler}
```
Acho que funciona.